src/mixins/java/org/spongepowered/common/mixin/core/world/level/LevelMixin.java

```diff
@@ -238,4 +243,29 @@ public abstract class LevelMixin implements LevelBridge, LevelAccessor {
 
         return (E) entity;
     }
+
+    @Inject(method = {
+        "getEntities(Lnet/minecraft/world/entity/Entity;Lnet/minecraft/world/phys/AABB;Ljava/util/function/Predicate;)Ljava/util/List;",
+        "getEntities(Lnet/minecraft/world/level/entity/EntityTypeTest;Lnet/minecraft/world/phys/AABB;Ljava/util/function/Predicate;)Ljava/util/List;",
+    }, at = @At("RETURN"))
+    private void impl$IgnoreTargetingOfVanishedEntities(
```

Comment:
This is a little drastic, and I may want to take advantage of the PhaseTracker to handle when vanishing logic comes into play (specifically for AI ticks, or overall).